core/src/main/scala/org/apache/spark/rdd/h2o/H2OAwareRDD.scala

```diff
@@ -14,20 +14,21 @@
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
-package org.apache.spark.listeners
+package org.apache.spark.rdd.h2o
 
-import org.apache.spark.scheduler.{SparkListener, SparkListenerExecutorAdded}
-import water.H2O
-import water.util.Log
+import org.apache.spark.rdd.RDD
+import org.apache.spark.{Partition, TaskContext}
 
-trait H2OSparkListener extends SparkListener {
-}
+import scala.reflect.ClassTag
 
-class ExecutorAddNotSupportedListener extends H2OSparkListener {
 
-  override def onExecutorAdded(executorAdded: SparkListenerExecutorAdded): Unit = {
-    Log.err(s"Executor ${executorAdded.executorId} without H2O instance discovered, killing the cloud!")
-    H2O.shutdown(-1)
-  }
+class H2OAwareRDD[U: ClassTag](nodes: Seq[String], prev: RDD[U]) extends RDD[U](prev: RDD[U]) {
+
+  override def getPreferredLocations(split: Partition): Seq[String] = nodes
```

Comment:
Does this method ensure that the logic will be always executed on the list of specified nodes? Aren't there any edge cases?

Reply:
Tests went fine, but still need to check the actual implementation, good questions

Reply:
More context -> It's not guaranteed for 100%. Spark will try to use the preferred locations, but only for duration of `spark.locality.wait` timeout. If the node preferred note is not available till then, it will try to launch the task on different executor.

We can set `spark.locality.wait=0` to ensure we will just try to use the right nodes. This does not affect as as we truly need to run tasks on nodes with H2O( and don't support dynamic allocation) not the other ones - so the behavior will still be the same as it is now

I think we will get rid of this anyway after the big stabilization effort.